WebaleGame.java

```diff
@@ -139,35 +139,50 @@ public boolean validMove(String type, int fromX, int fromY, int toX, int toY)
         else if(type.equals("Plus")){
             return true;
         }
-        
-        
-        ////Doesn't work T_T/////
+
         else if(type.equals("Triangle")){
-            if(x == y){
-                for(int i = 0; i < Math.abs(x); i++){
-                    if(toX < fromX){
-                        fromX = fromX - i;
-                    }
-                    else{
-                        fromX = fromX + i;
+            // triangle only can move diagonally, so abs(x) always equals to abs(y)
+            if(Math.abs(x) == Math.abs(y)){
+                //upper left
+                if(toX < fromX && toY > fromY){
+                    for(int i = 1; i <= x; i++){
```

Comment:
This line makes Triangle can't eat other pieces. It should be like this
```suggestion
                    for(int i = 1; i < x; i++){
```